content/en/blog/2022/apisix.md

```diff
@@ -0,0 +1,306 @@
+              }
+          }
+      },
+      "upstream": {
+          "type": "roundrobin",
+          "nodes": {
+              "127.0.0.1:7080": 1
+          }
+      }
+  }'
+  ```
+
+2. Modify the `./examples/demo/otel-collector-config.yaml` file to add the OTLP HTTP Receiver.
+
+   ```shell
+   receivers:
+    otlp:
+      protocols:
+        grpc:
+        http:${ip:port}   # add OTLP HTTP Receiver，default port is 4318
+
+    ```
+
+3. Modify `docker-compose.yaml` file.
+
+  You need to modify the configuration file, change the interface address of Client calling Server to the address of Apache APISIX, and map the ports of OTLP HTTP Receiver and Server services to local.
+
+  The following example is the complete docker-compose.yaml after the configuration is modified:
+
+  ```yaml
+  version: "2"
+  services:
+
+    # Jaeger
+    jaeger-all-in-one:
+      image: jaegertracing/all-in-one:latest
+      ports:
+        - "16686:16686" # jaeger ui port
+        - "14268"
+        - "14250"
+
+    # Zipkin
+    zipkin-all-in-one:
+      image: openzipkin/zipkin:latest
+      ports:
+        - "9411:9411"
+
+    # Collector
+    otel-collector:
+      image: ${OTELCOL_IMG}
+      command: ["--config=/etc/otel-collector-config.yaml", "${OTELCOL_ARGS}"]
+      volumes:
+        - ./otel-collector-config.yaml:/etc/otel-collector-config.yaml
+      ports:
+        - "1888:1888"   # pprof extension
+        - "8888:8888"   # Prometheus metrics exposed by the collector
+        - "8889:8889"   # Prometheus exporter metrics
+        - "13133:13133" # health_check extension
+        - "4317"        # OTLP gRPC receiver
+        - "4318:4318"   # Add OTLP HTTP Receiver port mapping
+        - "55670:55679" # zpages extension
+      depends_on:
+        - jaeger-all-in-one
+        - zipkin-all-in-one
+
+    demo-client:
+      build:
+        dockerfile: Dockerfile
+        context: ./client
+      environment:
+        - OTEL_EXPORTER_OTLP_ENDPOINT=otel-collector:4317
+        - DEMO_SERVER_ENDPOINT=http://172.17.0.1:9080/hello # APISIX address
+      depends_on:
+        - demo-server
+
+    demo-server:
+      build:
+        dockerfile: Dockerfile
+        context: ./server
+      environment:
+        - OTEL_EXPORTER_OTLP_ENDPOINT=otel-collector:4317
+      ports:
+        - "7080:7080" # Map the Server port to the host
+      depends_on:
+        - otel-collector
+
+    prometheus:
+      container_name: prometheus
+      image: prom/prometheus:latest
+      volumes:
+        - ./prometheus.yaml:/etc/prometheus/prometheus.yml
+      ports:
+        - "9090:9090"
+  ```
+
+It should be noted that demo-client.environment.DEMO_SERVER_ENDPOINT needs to be changed to your Apache APISIX address, and ensure that it can be accessed normally in the container.
+
+Of course, you can also deploy Apache APISIX through docekr-compose.yaml. For details, please refer to the Apache APISIX official documentation.
+
+### Step 3: Verify the Outputs
```

Comment:
either in this section or next one would be very important to add the following:

- With plugin you presumably get visibility into cases when the request does not reach the demo-server. Is there some kind of policies may be implemented on APISIX? What span will be generated in this case?
- Important to note, that even if demo-server is not instrumented, by enabling this plugin you are getting visibility into demo-server behavior. So it is a poor man substitute of a real instrumentation of demo-server which is extremely easy to do and out of the box provides a lot of value.

Reply:
Ditto, Haochao is working on it, should be delivered by this coming Tuesday.

Reply:
Updated.